lib/cylc/dbstatecheck.py

```diff
@@ -123,11 +131,12 @@ def task_state_met(self, task, cycle, status=None, message=None):
         if status:
             return bool(res)
         elif message:
-            for outputs_str, in res:
-                for value in json.loads(outputs_str).values():
-                    if message == value:
-                        return True
-            return False
+            # This works for Cylc 7 and Cylc 8 outputs table format:
+            return any(
+                message == value
+                for outputs_str, in res
+                for value in json.loads(outputs_str)
+            )
```

Comment:
Are you saying removing the `.values()` makes it work for Cylc 8 DBs? Surely that breaks Cylc 7?

Reply:
Or was that just a mistake?
```suggestion
            # This works for Cylc 7 and Cylc 8 outputs table format:
            return any(
                message == value
                for outputs_str, in res
                for value in json.loads(outputs_str).values()
            )
```

Reply:
Damn it. I owe you a chocolate fish 🐟 

It wasn't a mistake - I took it from master, and I tested it. 
However, I tested it with our classic shortcut `x = x` outputs - i.e. with message and label the same 🤦 

So this is a bug on master too. Patch coming...

Reply:
> So this is a bug on master too.

You mean, only a bug on master/8.x? (Though nearly introduced to Cylc 7 in this PR)